primitives/src/witness.rs

```diff
@@ -103,6 +103,7 @@ impl Witness {
     pub fn is_empty(&self) -> bool { self.witness_elements == 0 }
 
     /// Returns a struct implementing [`Iterator`].
+    #[must_use = "returned iterator should be used"]
```

Comment:
I believe there's a canonical message for iterators and it's not this. IIRC it mentions the word "lazy". If anyone is bored checking `std` and changing it would be nice.